mitxgraders/helpers/validatorfuncs.py

```diff
@@ -104,10 +104,10 @@ def all_unique(iterable):
     items should be unique, but have unexpected duplicates: ['a', 0]
     """
     seen = set()
-    duplicates = set()
+    duplicates = []
     for item in iterable:
         if item in seen:
-            duplicates.add(item)
+            duplicates.append(item)
```

Comment:
I think this doesn't work if you pass in ['a', 0, 0, 0], as 0 gets included in duplicates twice. That's why we were using a `set`.

Reply:
Probably means we need to add that case as a new test.

Reply:
The set version of duplicates wasn't good enough, because it is unordered and generally cannot be sorted by sorted (e.g., in the doctest above where strings and integers are incomparable for sorting).

So this version keeps duplicates sorted by order in the original iterable object.

Updated the test to have two copies of `0`